index.js

```diff
@@ -136,12 +136,23 @@ async function init() {
 	const repoListingConfig = {
 		user,
 		repository,
-		ref,
-		directory: decodeURIComponent(dir),
 		token: localStorage.token,
 		getFullData: true,
 	};
-	let files = await listContent.viaTreesApi(repoListingConfig);
+	let files;
+	const dirParts = decodeURIComponent(dir).split('/');
+
+	while (dirParts.length >= 0) {
+		try {
+			files = await listContent.viaTreesApi(repoListingConfig); // eslint-disable-line no-await-in-loop
+			break;
+		} catch {
+			ref += '/' + dirParts.shift();
+			repoListingConfig.directory = dirParts.join('/');
+			repoListingConfig.ref = ref;
+		}
+	}
```

Comment:
Can you extract this to its own function? It should `return await listContent.viaTreesApi(repoListingConfig)` as suggested